front-end/src/app/reports/f3x/submission-workflow/submit-f3x-step2.component.ts

```diff
@@ -0,0 +1,192 @@
+    'treasurer_last_name',
+    'treasurer_middle_name',
+    'treasurer_prefix',
+    'treasurer_suffix',
+    'filing_password',
+    'truth_statement',
+  ];
+  report: F3xSummary | undefined;
+  report_code = '';
+  stateOptions: PrimeOptions = [];
+  countryOptions: PrimeOptions = [];
+  formSubmitted = false;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  committeeAccount$: Observable<CommitteeAccount> = this.store.select(selectCommitteeAccount);
+  reportCodeLabelList$: Observable<ReportCodeLabelList> = new Observable<ReportCodeLabelList>();
+  form: FormGroup = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+  f3xReportCodeDetailedLabels = f3xReportCodeDetailedLabels;
+  loading: 0 | 1 | 2 = 0;
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private f3xSummaryService: F3xSummaryService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private store: Store,
+    private messageService: MessageService,
+    protected confirmationService: ConfirmationService
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
+    this.stateOptions = LabelUtils.getPrimeOptions(StatesCodeLabels);
+    this.countryOptions = LabelUtils.getPrimeOptions(CountryCodeLabels);
+
+    this.report = this.activatedRoute.snapshot.data['report'];
+    if (this.report?.report_code) {
+      if (this.report?.report_code?.length > 0) {
+        this.report_code = this.report.report_code;
+      }
+    } else {
+      this.report_code = '';
+    }
+    this.store
+      .select(selectCommitteeAccount)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((committeeAccount) => this.setDefaultFormValues(committeeAccount));
+
+    this.reportCodeLabelList$ = this.store.select<ReportCodeLabelList>(selectReportCodeLabelList);
+    this.store.dispatch(updateLabelLookupAction());
+
+    // Initialize validation tracking of current JSON schema and form data
+    this.validateService.formValidatorSchema = f3xSchema;
+    this.validateService.formValidatorForm = this.form;
+    this.form.controls['filing_password'].addValidators(Validators.required);
+    this.form.controls['truth_statement'].addValidators(Validators.requiredTrue);
+  }
+
+  setDefaultFormValues(committeeAccount: CommitteeAccount) {
+    this.form.patchValue({
+      //Take the required fields: first and last name
+      treasurer_last_name: this.report?.treasurer_last_name
+        ? this.report.treasurer_last_name
+        : committeeAccount?.treasurer_name_2,
+      treasurer_first_name: this.report?.treasurer_first_name
+        ? this.report.treasurer_first_name
+        : committeeAccount?.treasurer_name_1,
+    });
+
+    //If the report provided them, take the remaining fields from the report
+    if (this.report?.treasurer_last_name && this.report?.treasurer_first_name) {
```

Comment:
is there a reason that `treasurer_last_name` and `treasurer_first_name` are not handled with this condition?
It seems possible that we could end up with the first name from the report and last name from committee.  I don't think that's ever a desired outcome.  If we can assign first and last name the same way as the following fields, we could make this:

```
reportHasTreasurer = this.report?.treasurer_last_name && this.report?.treasurer_first_name;
treasurerSource = reportHasTreasurer ? this.report : this.committeeAccount;
this.form.patchValue({
  treasurer_last_name: treasurerSource.treasurer_last_name,
  treasurer_first_name: treasurerSource.treasurer_first_name,
  treasurer_middle_name: treasurerSource.treasurer_middle_name,
  treasurer_prefix: treasurerSource.treasurer_prefix,
  treasurer_suffix: treasurerSource.treasurer_suffix,
});
```

Reply:
Love the idea, used it for handling the addresses even, but it doesn't work here because the name attributes are named differently on the CommitteeAccount than they are on the Report. (e.g. "treasurer_first_name" to "treasurer_name_first")

Reply:
oh yeah that's a bummer!  what do you think about the split conditions for first and last names, though?  Do you think they need to be handled separately or can they be included in this if/else?

Reply:
As to the condition you describe where the treasurer_last_name comes from one source and the first name from another, this can only happen if you submit bad data to the API by going around the required fields on the front-end.  That said, I'll update this to avoid that possibility anyways.